source/extensions/filters/http/composite/action.h

```diff
@@ -43,29 +43,21 @@ class ExecuteFilterActionFactory
 
     Envoy::Http::FilterFactoryCb callback = nullptr;
 
-    // TODO(tyxia) Update the logic later to create the filter from the `factoryContext` first if it
-    // is present.
-    TRY_NEEDS_AUDIT {
-      if (context.server_factory_context_.has_value()) {
-        callback = factory.createFilterFactoryFromProtoWithServerContext(
-            *message, context.stat_prefix_, context.server_factory_context_.value());
-      }
-    }
-    END_TRY CATCH(EnvoyException & e, {
-      // First, we try to create the delegated filter creation callback from server factory context.
-      // If it failed (i.e., the corresponding filter doesn't support this method), we log this
-      // message and fallback to creating the filter from factory context.
-      ENVOY_LOG(trace,
-                absl::StrCat(e.what(), ", fallback to creating the filter from factory context."));
-    });
-
-    if (callback == nullptr) {
-      RELEASE_ASSERT(context.factory_context_.has_value(),
-                     "The factory context must exist here to create the delegated filter");
+    // First, try to create the filter factory creation function from factory context (if exists).
+    if (context.factory_context_.has_value()) {
       callback = factory.createFilterFactoryFromProto(*message, context.stat_prefix_,
                                                       context.factory_context_.value());
     }
 
+    // If above failed, try to create the filter factory creation function from server factory
+    // context (if exists).
+    if (callback == nullptr && context.server_factory_context_.has_value()) {
+      callback = factory.createFilterFactoryFromProtoWithServerContext(
+          *message, context.stat_prefix_, context.server_factory_context_.value());
+    }
+
+    RELEASE_ASSERT(callback != nullptr, "Failed to get filter factory creation function.");
```

Comment:
Can we throw here? Is this code running on the worker thread?

Reply:
done. I have to move this function to .cc file to throw exception as exception is not allowed in header filter in general.

PTAL, Thanks